tests/test_edge_cases.py

```diff
@@ -1296,6 +1296,13 @@ class Model(BaseModel):
     }
 
 
+def test_no_name_conflict_in_constructor():
+    class Model(BaseModel):
+        pass
+
+    assert Model(**{'__pydantic_self__': 4, 'self': 2})
```

Comment:
Can you add the field self and assert m.self == 2?

Reply:
done!